src/main/kotlin/org/rust/ide/annotator/fixes/ConvertToTyUsingTryFromTraitFix.kt

```diff
@@ -63,34 +63,34 @@ abstract class ConvertToTyUsingTryTraitAndUnpackFix(
     override fun addFromCall(rsPsiFactory: RsPsiFactory, startElement: RsExpr, fromCall: RsExpr) {
         val parentFnRetTy = findParentFnOrLambdaRetTy(startElement)
         when {
-            parentFnRetTy != null && isFnRetTyResultAndMatchErrTy(startElement, parentFnRetTy) ->
+            parentFnRetTy != null && isFnRetTyResultAndMatchErrTy(startElement, parentFnRetTy, errTy) ->
                 startElement.replace(rsPsiFactory.createTryExpression(fromCall))
             else -> startElement.replace(rsPsiFactory.createNoArgsMethodCall(fromCall, "unwrap"))
         }
     }
+}
 
-    private fun findParentFnOrLambdaRetTy(element: RsExpr): Ty? =
-        findParentFunctionOrLambdaRsRetType(element)?.typeReference?.type
-
-    private fun findParentFunctionOrLambdaRsRetType(element: RsExpr): RsRetType? {
-        var parent = element.parent
-        while (parent != null) {
-            when (parent) {
-                is RsFunction -> return parent.retType
-                is RsLambdaExpr -> return parent.retType
-                else -> parent = parent.parent
-            }
+private fun findParentFunctionOrLambdaRsRetType(element: RsExpr): RsRetType? {
+    var parent = element.parent
+    while (parent != null) {
+        when (parent) {
+            is RsFunction -> return parent.retType
+            is RsLambdaExpr -> return parent.retType
+            else -> parent = parent.parent
         }
-        return null
     }
+    return null
+}
 
-    private fun isFnRetTyResultAndMatchErrTy(element: RsExpr, fnRetTy: Ty): Boolean {
-        val items = element.knownItems
-        val lookup = ImplLookup(element.project, items)
-        return fnRetTy is TyAdt && fnRetTy.item == items.Result
-            && lookup.select(TraitRef(fnRetTy.typeArguments.get(1), (items.From
-            ?: return false).withSubst(errTy))).ok() != null
-    }
+fun findParentFnOrLambdaRetTy(element: RsExpr): Ty? =
```

Comment:
Maybe move these function to `org/rust/ide/utils/ExprUtils.kt`? Because it's not good to use functions from one annotator in another one.